src/network_messages/common_def.h

```diff
@@ -54,3 +57,17 @@ static inline bool operator!=(const IPv4Address& a, const IPv4Address& b)
 {
     return a.u32 != b.u32;
 }
+// Compute the siblings array of each level of tree
+template <unsigned int depth>
+static void getSiblings(int digestIndex, const m256i* digests, m256i siblings[depth])
+{
+    const unsigned int capacity = (1ULL << depth);
+    int siblingIndex = digestIndex;

```

Comment:
Do we need a lock here?
Same as the siblings for entity. Maybe while it updating `universe` & `spectrum`, these accesses would return wrong results. It's rare but it can happen.
please consider if we really need a lock here, (or users can just request again if the digest is mismatched).
If a lock is not needed, then maybe add some comments here about that race condition.